### ISSUE TITLE
MicrocalDataSet.phase_correct fails with too few "filled bins" of pulse height

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#156]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/156) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Both Malcolm and @pszypryt have noticed that `MicrocalDataSet.phase_correct(...)` fails if there are too few pulses in a channel. Specifically, it fails with the error 

```python

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/scipy/interpolate/fitpack2.pyc in __init__(self, x, y, w, bbox, k, s, ext, check_finite)
    183 
    184         data = dfitpack.fpcurf0(x,y,k,w=w,
--> 185                                 xb=bbox[0],xe=bbox[1],s=s)
    186         if data[-1] == 1:
    187             # nest too small, setting to maximum bound

error: (m>k) failed for hidden m: fpcurf0:m=2
> /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/scipy/interpolate/fitpack2.py(185)__init__()
    183 
    184         data = dfitpack.fpcurf0(x,y,k,w=w,
--> 185                                 xb=bbox[0],xe=bbox[1],s=s)
    186         if data[-1] == 1:
    187             # nest too small, setting to maximum bound
```

And also, I think I see why. It's because the spline in this line:

```
            crazy_spline = sp.interpolate.UnivariateSpline(x[nonempty], y[nonempty], w=w[nonempty]*(12**-0.5))
```

is being called with 3 or fewer nonempty bins. Let me create a test case and see where we can get.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Make phase_correct work even on small data sets.

Fixes #156 and adds a regression test. Bump active version to 0.6.4.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Merged in fowler (pull request #61)

Fix #156: phase correction on low-statistics data sets

Approved-by: Joseph Fowler <drjoefowler@gmail.com>

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
